src/contracts/core/DelegationManager.sol

```diff
@@ -728,61 +702,84 @@ contract DelegationManager is
+            completableTimestamp = startTimestamp + MIN_WITHDRAWAL_DELAY;
         }
     }
 
-    function _setMinWithdrawalDelayBlocks(uint256 _minWithdrawalDelayBlocks) internal {
-        require(
-            _minWithdrawalDelayBlocks <= MAX_WITHDRAWAL_DELAY_BLOCKS,
-            "DelegationManager._setMinWithdrawalDelayBlocks: _minWithdrawalDelayBlocks cannot be > MAX_WITHDRAWAL_DELAY_BLOCKS"
-        );
-        emit MinWithdrawalDelayBlocksSet(minWithdrawalDelayBlocks, _minWithdrawalDelayBlocks);
-        minWithdrawalDelayBlocks = _minWithdrawalDelayBlocks;
-    }
-
     /**
-     * @notice Sets the withdrawal delay blocks for each strategy in `_strategies` to `_withdrawalDelayBlocks`.
-     * gets called when initializing contract or by calling `setStrategyWithdrawalDelayBlocks`
+     *
+     *                              SHARES CONVERSION FUNCTIONS
+     *
      */
-    function _setStrategyWithdrawalDelayBlocks(
-        IStrategy[] calldata _strategies,
-        uint256[] calldata _withdrawalDelayBlocks
-    ) internal {
-        require(
-            _strategies.length == _withdrawalDelayBlocks.length,
-            "DelegationManager._setStrategyWithdrawalDelayBlocks: input length mismatch"
-        );
-        uint256 numStrats = _strategies.length;
-        for (uint256 i = 0; i < numStrats; ++i) {
-            IStrategy strategy = _strategies[i];
-            uint256 prevStrategyWithdrawalDelayBlocks = strategyWithdrawalDelayBlocks[strategy];
-            uint256 newStrategyWithdrawalDelayBlocks = _withdrawalDelayBlocks[i];
-            require(
-                newStrategyWithdrawalDelayBlocks <= MAX_WITHDRAWAL_DELAY_BLOCKS,
-                "DelegationManager._setStrategyWithdrawalDelayBlocks: _withdrawalDelayBlocks cannot be > MAX_WITHDRAWAL_DELAY_BLOCKS"
-            );
 
-            // set the new withdrawal delay blocks
-            strategyWithdrawalDelayBlocks[strategy] = newStrategyWithdrawalDelayBlocks;
-            emit StrategyWithdrawalDelayBlocksSet(
-                strategy, prevStrategyWithdrawalDelayBlocks, newStrategyWithdrawalDelayBlocks
+    /**
+     * @notice helper to calculate the new depositScalingFactor after adding shares. This is only used
+     * when a staker is depositing through the StrategyManager or EigenPodManager. A staker's depositScalingFactor
+     * is only updated when they have new deposits and their shares are being increased.
+     */
+    function _updateDepositScalingFactor(
+        address staker,
+        IStrategy strategy,
+        uint64 totalMagnitude,
+        uint256 existingPrincipalShares,
+        uint256 addedShares
+    ) internal returns (uint256) {
+        uint256 newDepositScalingFactor;
+
+        if (existingPrincipalShares == 0) {
+            // existing shares are 0, meaning no existing delegated shares. In this case, the new depositScalingFactor
+            // is re-initialized to
+            newDepositScalingFactor = WAD / totalMagnitude;
```

Comment:
**ALERT -- HIGHLY PROBABLE BUG:** there is going to be brutal loss of precision here as this is currently implemented. Right now `totalMagnitude` is initialized as equal to `WAD` (from consulting AllocationManager code), so this calculation will return `1` until `totalMagnitude <= WAD / 2` -- this certainly doesn't seem intended!!

Reply:
Nice catch, this is something we've discussed. WIP.